### DISCUSSION TITLE
How can I get upload files?

### DISCUSSION THREAD

**Discussion by MarkAntipin** [fastapi/fastapi#8071] [Questions] [answered]
**Description**

How can I get upload files?
Is it possible to do it like in flask?

**Additional context**
In docs you have:

```python
@app.post("/")
async def some_end(files: List[UploadFile] = File(...)):
    return [file.file for file in files]

```
but, i want to pass files with 'keys' like file1, file2...
In flask i can just write 

```python
from flask import request

for f in request.files:
    pass
```
So how can i do it in fastAPI?

**Comment by euri10** [comment]
:flushed: 
not clear what you mean, if I understood correctly what you want, in Flask the doc says :

> The application accesses the file from the files dictionary on the request object

So `request.files` in Flask would be kind of equivalent to the `files` parameter in your FastAPI endpoint above, the difference being `files` is a list of all the files which are typed as `UploadFile`, while the other is a dictionary.  

Now you can loop through them, get their name, length, type, well whatever you want to do with it, the same way you would do in Flask.

**Comment by MarkAntipin** [comment]
In flask if i pass a dict of files {'file1': 'file_ob1', file2': 'file_ob2, ...}

```python
for f in request.files:
    print(f)
```
i will get file1, file2, ...

in fast api i need to specify specify
```python
async def some_end(file1: UploadFile = File(...), file2: UploadFile = File(...)):
    pass
```
So i want to pass files without specifying keys
Sorry if i'm not clear enough

**Comment by euri10** [comment]
I don't think so it's a list of files you get, can't confirm for sure, but
I will do later
Maybe check
https://fastapi.tiangolo.com/tutorial/request-files/#multiple-file-uploads

Le ven. 13 sept. 2019 à 9:28 PM, MarkAntipin <notifications@github.com> a
écrit :

**Comment by MarkAntipin** [comment]
Ok, thank you anyway

**Comment by tiangolo** [comment, marked as answer]
Thanks for the help here @euri10 ! :clap: :bow: 

Thanks for reporting back and closing the issue :+1: 

---

@MarkAntipin you could use the request directly and extract the files, but you would lose the OpenAPI generation, UI docs, etc. https://fastapi.tiangolo.com/advanced/using-request-directly/
